### ISSUE TITLE
could not output training message when training

### ISSUE THREAD

**Issue by aobuke** [dmlc/xgboost#2089]
For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

update:
my IDE is Spyder, it could print message on terminal while not on my Spyder, verbose_eval is not work

## Environment info
Operating System:
Ubuntu 16.04.1 LTS 64bit
Compiler:
5.4.0 20160609
Package used (python/R/jvm/C++):
python 2.7.11 Anaconda 4.1.0 64bit
`xgboost` version used:
0.6a2
If you are using python package, please provide

1. The python version and distribution
2. The command to install `xgboost` if you are not installing from source
pip install xgboost

## Steps to reproduce

1. param{'objective':'reg:linear','booster':'gbtree','silent':0}
2.
3.

## What have you tried?

1. 'silent':False
2. xgb.train(param,dtrain, verbose_eval = True)
3. downgrade to xgb version 0.4

**Comment by khotilov** [comment]
xgb.train?

 evals: list of pairs (DMatrix, string)
        List of items to be evaluated during training, this allows user to watch
        performance on the validation set.
